### PR TITLE
Ask to reset session if authenticated member tries to edit SSN in purchase form

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Warn about changing SSN if member is authenticated.  If member decides to proceed, reset session and reload so that member sees editable SSN field

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-2193]: https://hedvig.atlassian.net/browse/GRW-2193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1598** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1598" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1598?utm_source=stack-comment).

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx [hunk 1]
```diff
@@ -88,12 +90,53 @@ const NewMemberSsnSection = ({ shopSession, onCompleted }: Props) => {
...
+
+const ResetSessionButton = () => {
   const { t } = useTranslation('purchase-form')
+  const refreshRouter = useRouterRefresh()
+  const [loading, setLoading] = useState(false)
+  const { reset: resetShopSession } = useShopSession()
+  const handleClick = async () => {
+    setLoading(true)
+    await resetShopSession()
+    await refreshRouter()
```
Comment (robinandeer) [inline apps/store/src/components/PriceCalculator/SsnSeSection.tsx:128]: Should we do a full page reload just to be safe?

File: apps/store/src/services/authApi/persist.ts [hunk 2]
```diff
@@ -13,18 +13,18 @@ const OPTIONS: OptionsType = {
   }),
 }
 
-export const saveAccessToken = (accessToken: string, { req, res }: CookieParams = {}) => {
-  setCookie(COOKIE_KEY, serialize(accessToken), { req, res, ...OPTIONS })
+export const saveAccessToken = (accessToken: string, cookieParams: CookieParams) => {
```
Comment (robinandeer) [inline apps/store/src/services/authApi/persist.ts:16]: Are we ever calling this server side? Do we need cookie params

Can we make the second argument optional -- looks a big strange to always pass an empty object

File: apps/store/src/services/shopSession/ShopSessionContext.tsx [hunk 3]
```diff
@@ -72,6 +74,14 @@ const useShopSessionContextValue = (initialShopSessionId?: string) => {
     [queryResult.shopSession],
   )
 
+  queryResult.reset = useCallback(() => {
+    shopSessionServiceClientSide.reset()
+    resetAccessToken({})
+    return shopSessionServiceClientSide.getOrCreate({ countryCode }).then((shopSession) => {
```
Comment (robinandeer) [inline apps/store/src/services/shopSession/ShopSessionContext.tsx:80]: isn't there a create endpoint? feels strange to reset and then "get or create" 🤔

**Review by robinandeer — APPROVED** [review]
Nice work -- just added some quality of life improvement ideas

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/components/PriceCalculator/SsnSeSection.tsx:128]: This is an alternative, but we'd need to reopen price calculator based on URL - let's build it later

File: apps/store/src/services/shopSession/ShopSessionContext.tsx (hunk 3 above)
Comment (alebedev) [reply apps/store/src/services/shopSession/ShopSessionContext.tsx:80]: It'll need some modifications to support deduplication of requests.  Let's see how simple that would be

**Comment by alebedev** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: BpIRi7Ft7erGYmEC6w5s.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1598)

**Pull request merged**